### PR TITLE
Object meta fixes in federation controllers

### PR DESCRIPTION
**Pull request by mwielgus**
This PR adds helper functions for handling object meta structs and applies them to the existing controllers.

cc: @quinton-hoole @jianhuiz @kubernetes/sig-cluster-federation

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31076)

### CODE REVIEW

File: federation/pkg/federation-controller/util/meta.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+package util
+
+import (
+	"reflect"
+
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+func CopyObjectMeta(obj api_v1.ObjectMeta) api_v1.ObjectMeta {
```
Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta.go]: Add a doc comment please.

File: federation/pkg/federation-controller/util/meta.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	return api_v1.ObjectMeta{
+		Name:        obj.Name,
+		Namespace:   obj.Namespace,
+		Labels:      obj.Labels,
+		Annotations: obj.Annotations,
+	}
+}
+
+func ObjectMetaEquivalent(a, b api_v1.ObjectMeta) bool {
```
Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta.go]: Add a doc comment please.

Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta.go]: You need to explain why you chose these fields, and not others, and what needs to be done if fields are added to ObjectMeta in future.

File: federation/pkg/federation-controller/util/meta_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		UID:         "1231231412",
+		Annotations: map[string]string{"A": "B"},
+	}
+
+	o2 := CopyObjectMeta(o1)
+	assert.Equal(t, 0, len(o2.UID))
+	assert.Equal(t, 0, len(o2.ResourceVersion))
+	assert.Equal(t, o1.Name, o2.Name)
+	assert.True(t, ObjectMetaEquivalent(o1, o2))
+	assert.False(t, ObjectMetaEquivalent(o1, o3))
```
Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta_test.go]: You need a more comprehensive test here, more along the lines of https://github.com/kubernetes/kubernetes/blob/master/federation/pkg/dnsprovider/dns_test.go#L89

**Comment by quinton-hoole** [comment]
Thanks @mwielgus .  LGTM barring some nits that can be addressed in a followup PR.

File: federation/pkg/federation-controller/util/meta.go (hunk 1 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/util/meta.go]: Added.

File: federation/pkg/federation-controller/util/meta.go (hunk 2 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/util/meta.go]: Added.

File: federation/pkg/federation-controller/util/meta_test.go (hunk 3 above)
Comment (mwielgus) [inline federation/pkg/federation-controller/util/meta_test.go]: Done.

File: federation/pkg/federation-controller/util/meta.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+package util
+
+import (
+	"reflect"
+
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+// Copies cluster-independent, user provided data from the given ObjectMeta struct. If in
```
Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta.go]: @mwielgus Please start with function name, as per godoc conventions: https://blog.golang.org/godoc-documenting-go-code

File: federation/pkg/federation-controller/util/meta.go [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+func CopyObjectMeta(obj api_v1.ObjectMeta) api_v1.ObjectMeta {
+	return api_v1.ObjectMeta{
+		Name:        obj.Name,
+		Namespace:   obj.Namespace,
+		Labels:      obj.Labels,
+		Annotations: obj.Annotations,
+	}
+}
+
+// Checks if cluster-independed, user provided data in two given ObjectMeta are eqaul. If in
```
Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta.go]: Please start with function name, as per godoc conventions: https://blog.golang.org/godoc-documenting-go-code

File: federation/pkg/federation-controller/util/meta_test.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import (
+	"testing"
+
+	api_v1 "k8s.io/kubernetes/pkg/api/v1"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestObjectMeta(t *testing.T) {
+	o1 := api_v1.ObjectMeta{
```
Comment (quinton-hoole) [inline federation/pkg/federation-controller/util/meta_test.go]: @mwielgus Please use table driven testing as per Kubernetes conventions:

https://github.com/kubernetes/kubernetes/blob/master/docs/devel/coding-conventions.md#testing-conventions

and as per example that I sent in previous PR comment:

https://github.com/kubernetes/kubernetes/blob/master/federation/pkg/dnsprovider/dns_test.go#L53

**Pull request merged**